### DISCUSSION TITLE
DIPY lab meetings, Spring 2020

### DISCUSSION THREAD

**Discussion by arokem** [dipy/dipy#2355] [2. Announcements]
The schedule for upcoming lab meetings is:

April 9th, 0800 PT, 1100 ET, 1600 UTC, Topic: Overview of the release 1.2 (@skoudoro)
April 23rd, 0800 PT, 1100 ET, 1600 UTC Topic:  Deep learning reveals untapped information for local white-matter fiber reconstruction in diffusion-weighted MRI (@finalement)
May 7th, 0800 PT, 1100 ET, 1600 UTC Topic: ???
May 21st, 0800 PT, 1100 ET, 1600 UTC Topic: GSoC Student presentation and future work (@ShreyasFadnavis)

All meetings will be held through Zoom at the following link https://iu.zoom.us/j/345602333

Meetings will be 60 minutes at the longest.

**Comment by arokem** [comment]
OK - we have a schedule! But we need topics... 

Does anyone want to volunteer to lead one of the upcoming meetings?

**Comment by skoudoro** [comment]
I think the upcoming one should start with the release since it is coming for April 13th. I will lead that one and update the schedule.

**Comment by skoudoro** [comment]
Here some ideas/propositions:

- it would be good if @StongeEtienne could lead a meeting about mesh/surface. What do you think @StongeEtienne? 
- it would be good to have a Q/A or AMA with our user and promote it
- The last spot (May 21st) could be a GSOC introduction, with a reminder about the goal of machine learning + image registration. @ShreyasFadnavis could lead that one with me

Any other suggestions are more than welcome!

**Comment by skoudoro** [comment]
A short reminder for the meeting today in 20min: 0800 PT, 1100 ET, 1600 UTC.

see you all!

**Comment by skoudoro** [comment]
Hi all,

Just a small reminder: tomorrow, Thursday, April 23rd (0800 PT, 1100 ET, 1600 UTC), There is the presentation of @finalelement during the DIPY online meeting. 

Topic: Deep learning reveals untapped information for local white-matter fiber reconstruction in diffusion-weighted MRI. Magnetic resonance imaging, 62, 220-227.
link:  https://meet.jit.si//dipy-lab-meeting

See you tomorrow @dipy/dipy-community

**Comment by finalelement** [comment]
Hi Serge,

Thank you for the friendly reminder!

Cheers,
Vish

**Comment by ShreyasFadnavis** [comment]
Hi all,

@sarathknv who will be participating in GSoC + DIPY this year, would like to present his work in the DIPY lab meeting. He will be implementing a `deep learning` based `2D and 3D registration` as a part of this. He just notified me saying that he has some end-of-semester tasks to finish at school and would like to postpone this presentation by a week... Is that okay with everyone? 

Thanks,
Shreyas

**Comment by Garyfallidis** [comment]
That is fine with me.

Can I also suggest that @grlee77 gives a presentation at his convenience? @grlee77 is working on two awesome projects. GPU acceleration for DIPY and wavelet-based denoising. @grlee77 be happy to decide which project you would like to talk about first and let us know about potential dates that work for you.

**Comment by grlee77** [comment]
I will try to join in and watch @sarathknv's presentation this week.

Let me know what the schedule for the coming weeks is like. Whatever slot is convenient will probably work for me. I would prefer to present some preliminary work on GPU-based acceleration for DIPY first and then follow up in a later slot regarding some recent denoising work.

**Comment by skoudoro** [comment]
Hey @dipy/dipy-community,

The DIPY online meeting is canceled for tomorrow (many people can not attend). The future talk are:

- May 28st, 0800 PT, 1100 ET, 1600 UTC Topic: Deep Learning and image registration by GSoC student(@sarathknv) Data Generators for DL by @finalelement  (30min)

- June 11th, 0800 PT, 1100 ET, 1600 UTC Topic: GPU-based acceleration for DIPY by @grlee77 


See you online next week. 

Serge K.
ps: I might create a new issue and close this one later

**Comment by Garyfallidis** [comment]
Jit.si seems performing worse than Zoom especially concerning sound quality. @arokem for next session let's move back to zoom.

**Comment by skoudoro** [comment]
Hello @dipy/dipy-community,

A short reminder for the talk of the day (June 11th, 0800 PT, 1100 ET, 1600 UTC) by @grlee77: **GPU-based acceleration for DIPY** 

Link: https://meet.jit.si//dipy-lab-meeting

See you online!

Serge K.

**Comment by Garyfallidis** [comment]
Given some issues with jitsi we are moving back to zoom.
Here is the link https://iu.zoom.us/j/345602333

**Comment by Garyfallidis** [comment]
Make sure you join at this address from today and onward 
https://iu.zoom.us/j/345602333
Thank you all!

**Comment by grlee77** [comment]
Here are the slides and notebooks from yesterday's talk. I transferred the notebooks to Google Colab so you can just run them in the browser using GPUs in the cloud. Otherwise it can be time consuming to get all of the GPU computing frameworks that were involved set up in the same local environment.

I typically get slower results on Colab than on my local workstation, mainly because you get whatever GPU is available and it is often not from the most recent generation. Can't complain too much about that, though, as it is free!

I believe I have set the permissions so that anyone with the links can view them, but please let me know if you have any issues accessing the materials.

**Presentation Slides:**
https://drive.google.com/file/d/1_tiLSVvRcengG-Vao73zYlPWTlUpr-yE/view?usp=sharing

**CuPy: NEP13/18 demo and GPU tensor library interoperability:**
https://colab.research.google.com/drive/1NGOBcRpAYXlPDToGxdVhzmnvXB-3J-K3?usp=sharing

**Benchmarking some ndimage functions on GPU vs CPU:**
https://colab.research.google.com/drive/1dJFe3VA2Qh2j6uDNmuZop-VKXNhiFuVy?usp=sharing

**Examples converting two fairly simple functions for use on the GPU for three different libraries implementing the NumPy API (CuPy, JAX, MXNet)**
https://colab.research.google.com/drive/1AOrjmV16SS54DNh0suZIzWKEiWfmHYDQ?usp=sharing

The only thing not public yet is the DIPY-specific GPU code.

**Comment by skoudoro** [comment]
Hi @arokem,

I think it is time to start the new season! DIPY meeting FALL 2020. 

Can you create a new issue?

**Comment by skoudoro** [comment]
Superseded by #2229

**Comment by grlee77** [comment]
The GPU implementations covered in my June 2020 presentation are now up at https://github.com/dipy/cudipy. Please try it out and let me know if you encounter any issues. The specific functions covered are highlighted in the README and by the adaptation of previously existing DIPY examples in `doc/examples`.

I did manage to make another ~4x speed improvement to MRF-based tissue classification from what was presented in June. The demo from `doc/examples/tissue_classification.py` now does the segmentation in only ~2 seconds on a GTX 1080 Ti which is an acceleration factor of around 50 from the CPU-based case.

Another factor of ~2x improvement to Gibb artifact removal was also made. This function has also been substantially accelerated on the CPU by recent PRs (#2250, #2253, #2265). The parallelization on the CPU was very effective with a speedup roughly equal to the number of cores. The GPU-based implementation is now only approximately 2x faster than a 10-core parallel CPU implementation.
